test/unittests/eof_validation.cpp

```diff
@@ -110,9 +110,17 @@ void eof_validation::TearDown()
     for (size_t i = 0; i < test_cases.size(); ++i)
     {
         const auto& test_case = test_cases[i];
-        EXPECT_EQ(evmone::validate_eof(rev, test_case.kind, test_case.container), test_case.error)
+
+        // Move the container to new heap-allocated buffer of exact size
```

Comment:
Wouldn't `test_case.container.shrink_to_fit()` achieve the same?

Reply:
Not for `std::string` where there is SSO and additional NULL terminator.